core/templates/domain/question/editable-question-backend-api.service.ts

```diff
@@ -41,13 +41,24 @@ angular.module('oppia').factory('EditableQuestionBackendApiService', [
       QUESTION_SKILL_LINK_URL_TEMPLATE) {
     var _createQuestion = function(
         skillIds, skillDifficulties,
-        questionDict, successCallback, errorCallback) {
+        questionDict, imagesData, successCallback, errorCallback) {
       var postData = {
         question_dict: questionDict,
         skill_ids: skillIds,
         skill_difficulties: skillDifficulties
       };
-      $http.post(QUESTION_CREATION_URL, postData).then(function(response) {
+      let body = new FormData();
+      body.append('payload', JSON.stringify(postData));
+      let filenames = imagesData.map(obj => obj.filename);
+      let imageBlobs = imagesData.map(obj => obj.imageBlob);
+      for (let idx in imageBlobs) {
+        body.append(filenames[idx], imageBlobs[idx]);
+      }
+      $http.post(QUESTION_CREATION_URL, (body), {
+        headers: {
+          'Content-Type': undefined
```

Comment:
Why are we doing this?

Reply:
I've added a docstring explaining the reason.
The actual header to be added for form-data is 'multipart/form-data',  But adding it manually won't work because we will miss the boundary parameter. When we keep 'Content-Type' as undefined, the browser automatically fills the boundary parameter according to the form data. Refer [link](https://stackoverflow.com/questions/37039852/. ) and
      [link](https://stackoverflow.com/questions/34983071/.)
Also this can be removed after we migrate to angular as only in angularjs $http we face this issue, in angular this is done automatically.
refer to  `_createSkill()` in `SkillCreationBackendApiService` which does the same process in angular